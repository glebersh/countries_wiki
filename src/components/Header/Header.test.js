import { fireEvent, render, screen } from '@testing-library/react';
import { toBeInTheDocument, toHaveBeenCalledWith } from '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { Header } from './Header';
import * as reduxHooks from 'react-redux';
import { BrowserRouter } from 'react-router-dom';


jest.mock('react-redux');
const mockUseDispatch = jest.spyOn(reduxHooks, 'useDispatch');
const mockUseSelector = jest.spyOn(reduxHooks, 'useSelector');


describe('Header component', () => {
  it('Header Link renders', () => {
    mockUseDispatch.mockReturnValue(jest.fn());

    render(<BrowserRouter>
      <Header />
    </BrowserRouter>);
    expect(screen.getByRole('link')).toBeInTheDocument();
  });

  it('Header snapshot', () => {
    const header = render(<BrowserRouter>
      <Header />
    </BrowserRouter>);
    expect(header).toMatchSnapshot();
  });

  it('Header Link dispatch triggering', () => {
    const dispatch = jest.fn();
    mockUseDispatch.mockReturnValue(dispatch);

    render(<BrowserRouter>
      <Header />
    </BrowserRouter>);
    userEvent.click(screen.getByRole('link'));
    expect(dispatch).toHaveBeenCalled();
  });
});
