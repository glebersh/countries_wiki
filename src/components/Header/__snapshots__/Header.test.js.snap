// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    data-theme="undefined"
  >
    <div>
      <header
        class="sc-dkrFOg kfwkcy"
      >
        <div
          class="sc-bcXHqe cQxGuT"
        >
          <div
            class="sc-hLBbgP eiRunD"
          >
            <a
              class="sc-eDvSVe kMSkcy"
              href="/countries_wiki"
            >
              Where is the world?
            </a>
            <div
              class="sc-gswNZR fDqBWa"
              data-testid="theme-switcher"
            >
              <svg
                fill="currentColor"
                height="14px"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="14px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M264 480A232 232 0 0132 248c0-94 54-178.28 137.61-214.67a16 16 0 0121.06 21.06C181.07 76.43 176 104.66 176 136c0 110.28 89.72 200 200 200 31.34 0 59.57-5.07 81.61-14.67a16 16 0 0121.06 21.06C442.28 426 358 480 264 480z"
                />
              </svg>
               
              <span
                style="margin-left: 0.75rem;"
              >
                 Theme
              </span>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-dkrFOg kfwkcy"
    >
      <div
        class="sc-bcXHqe cQxGuT"
      >
        <div
          class="sc-hLBbgP eiRunD"
        >
          <a
            class="sc-eDvSVe kMSkcy"
            href="/countries_wiki"
          >
            Where is the world?
          </a>
          <div
            class="sc-gswNZR fDqBWa"
            data-testid="theme-switcher"
          >
            <svg
              fill="currentColor"
              height="14px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="14px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M264 480A232 232 0 0132 248c0-94 54-178.28 137.61-214.67a16 16 0 0121.06 21.06C181.07 76.43 176 104.66 176 136c0 110.28 89.72 200 200 200 31.34 0 59.57-5.07 81.61-14.67a16 16 0 0121.06 21.06C442.28 426 358 480 264 480z"
              />
            </svg>
             
            <span
              style="margin-left: 0.75rem;"
            >
               Theme
            </span>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
